streams/src/test/java/org/apache/kafka/streams/state/internals/FilteredCacheIteratorTest.java

```diff
@@ -127,4 +132,56 @@ public void shouldThrowUnsupportedOperationExeceptionOnRemove() {
         allIterator.remove();
     }
 
+    public static class KeyValueStoreWrapper<K extends Comparable<K>, V> {
+        private final NavigableMap<K, V> map;
+
+        KeyValueStoreWrapper() {
+            map = new TreeMap<>();
+        }
+
+        private void putAll(final List<KeyValue<K, V>> entries) {
+            for (final KeyValue<K, V> entry : entries) {
+                map.put(entry.key, entry.value);
+            }
+        }
+
+        private KeyValueIterator<K, V> all() {
+            return new KeyValueIteratorWrapper<>(map.entrySet().iterator());
+        }
+
+        private static class KeyValueIteratorWrapper<K, V> implements KeyValueIterator<K, V> {
+            private final Iterator<Entry<K, V>> iter;
+
+            private KeyValueIteratorWrapper(final Iterator<Map.Entry<K, V>> iter) {
+                this.iter = iter;
+            }
+
+            @Override
+            public boolean hasNext() {
+                return iter.hasNext();
+            }
+
+            @Override
+            public KeyValue<K, V> next() {
+                final Map.Entry<K, V> entry = iter.next();
+                return new KeyValue<>(entry.getKey(), entry.getValue());
+            }
+
+            @Override
+            public void remove() {
+                iter.remove();
+            }
+
+            @Override
+            public void close() {
+                // do nothing
+            }
+
+            @Override
+            public K peekNextKey() {
+                throw new UnsupportedOperationException("peekNextKey() not supported in " + getClass().getName());
```

Comment:
Why not call `underlying` ? A "wrapper" should be agnostic if the wrapper iterator support `peekNextKey()` or not.

Reply:
It's just a wrapper around a basic Java Iterator which doesn't support peekNextKey()